### DISCUSSION TITLE
Attempt to read property "state" on null when custom transition

### DISCUSSION THREAD

**Discussion by imClement** [spatie/laravel-model-states#211] [Bugs]
Hi,

When using `$payment->state->transitionTo(Failed::class)` with a custom transition class defined in  config(), I get the following error:

``` 
Attempt to read property "state" on null
```

However, nor $payment or $payment->state are null (checked with Ray) 

No errors when  using `$payment->state->transitionTo(Failed::class)` without having a custom transition in config().

Any suggestion for debugging?
Thank you,

**Comment by imClement** [comment]
Trying to debug, I found the following.

For unknown reasons, the transitionTo() is executed twice when a custom transition class is defined. 

```
    public function transitionTo($newState, ...$transitionArgs)
    {

        $newState = $this->resolveStateObject($newState);
```

ref: vendor/spatie/laravel-model-states/src/State.php

**Comment by christianschoenmakers** [comment]
Same issue here, but it does work in a different project. Haven't figured out why yet. 

Tested on both versions 2.4.6 and 2.6.0.

**Comment by christianschoenmakers** [comment]
I figured out what was the issue on my end. In the `handle` method of the Transition class, I wasn't returning anything (void). After changing it to returning the Model instance, it did work. 

The example in the documentation is also returning the `Payment` model: https://spatie.be/docs/laravel-model-states/v2/working-with-transitions/02-custom-transition-classes
